cmd/plugin/commands/routes/routes.go

```diff
@@ -0,0 +1,129 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package routes
+
+import (
+	"strings"
+
+	"github.com/apache/apisix-ingress-controller/cmd/plugin/common"
+	"github.com/apache/apisix-ingress-controller/cmd/plugin/kubectl"
+	"github.com/apache/apisix-ingress-controller/pkg/log"
+	jsoniter "github.com/json-iterator/go"
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+)
+
+func CreateCommand(flags *genericclioptions.ConfigFlags) *cobra.Command {
+	var routeId string
+	cmd := &cobra.Command{
+		Use:   "routes",
+		Short: "Show the apisix routes",
+		Run: func(cmd *cobra.Command, args []string) {
+			getRoutes(flags, routeId)
+			return
+		},
+	}
+	cmd.Flags().StringVar(&routeId, "route-id", "", "apisix route id")
+	return cmd
+}
+
+// getRoutes get apisix route informations.
+func getRoutes(flags *genericclioptions.ConfigFlags, routeId string) {
+	pconf := common.NewPluginConfig(flags)
+	log.Infof("namespaces: %v \n", pconf.NameSpace)
+	pconf.GetConfigMapsData()
+
+	svcName, _ := pconf.GetApisixSvcName()
+	pid := kubectl.OpenPortForward(pconf.Ctx, pconf.NameSpace, svcName)
+	header := []string{
+		"ID",
+		"Name",
+		"Host",
+		"URI",
+		"Status",
+		"UpstreamId",
+		"CreateTime",
+		"UpdateTime",
+	}
+
+	for {
+		if common.CheckPort() {
+			log.Info("the k8s port-forward is ready")
+			break
+		}
+		log.Error("the k8s port-forward is not ready")
+	}
+	var reqApisix common.RequestConf
+	reqApisix.AdminKey = pconf.Cfg.APISIX.AdminKey
+	reqApisix.URL = "http://127.0.0.1:9180/apisix/admin"
```

Comment:
Cannot assume the APISIX address is just `127.0.0.1:9180`.

Reply:
Beacuse I used the kubernetes port-forward feature in the machine on which this command is run. So just visit the `127.0.0.1:9180`

Reply:
It's not reasonable to specify the port, we cannot sure whether this port is already in use.